tests/testthat/test-read_redcap_tidy.R

```diff
@@ -254,3 +254,61 @@ test_that("get_fields_to_drop handles checkboxes", {
   )
 })
 
+test_that("read_redcap_tidy returns metadata", {
+  httptest::with_mock_api({
+    out <- read_redcap_tidy(redcap_uri, longitudinal_token)
+  })
+
+  expected_cols <- c(
+    "redcap_form_name", "redcap_form_label", "redcap_data", "redcap_metadata",
+    "redcap_events", "structure", "data_rows", "data_cols", "data_size",
+    "data_na_pct"
+  )
+
+  # metadata fields exist and correctly ordered
+  expect_equal(expected_cols, names(out))
+
+  # metadata fields have the correct data types
+
+  ## redcap_metadata and redcap_events fields consist of tibbles
+  c(out$redcap_metadata, out$redcap_events) %>%
+    purrr::map_lgl(inherits, what = "tbl") %>%
+    all() %>%
+    expect_true()
+
+  ## summary fields have correct types
+  expect_type(out$data_rows, "integer")
+  expect_type(out$data_cols, "integer")
+  expect_s3_class(out$data_size, "lobstr_bytes")
+  expect_true(
+    all(out$data_na_pct >= 0) && all(out$data_na_pct <= 1)
+  )
+
+  # metadata contains all expected fields in data
+
+  ## fields we don't expect
+  exclude_fields <- c(
+    "redcap_repeat_instance", "redcap_event",
+    "redcap_arm", "form_status_complete"
+  )
+
+  fields_in_metadata <- out$redcap_metadata %>%
+    map("field_name") %>%
```

Comment:
This (ll. 295-301) is hard to understand. Can you rewrite w/o `map` functions?

Reply:
I'm not sure I can get away without mapping because this check tests that data fields and metadata fields match *within* each instrument (i.e. row of supertibble). I tried to clean things up and add clearer comments here: https://github.com/CHOP-CGTDataOps/REDCapTidieR/pull/58/commits/78ce9505122fe21325589e56c0fdd2236e352310. Let me know if that looks okay

Reply:
I think it looks good!